tests/components/zha/common.py

```diff
@@ -237,3 +238,10 @@ async def async_test_rejoin(hass, zigpy_device, clusters, report_counts, ep_id=1
         assert cluster.bind.await_count == 1
         assert cluster.configure_reporting.call_count == reports
         assert cluster.configure_reporting.await_count == reports
+
+
+async def async_wait_for_updates(hass):
+    """Wait until all scheduled updates are executed."""
+    await asyncio.sleep(0)
```

Comment:
why 2 of these back to back? this seems weird to me. is there something flaky in the tests that this is covering up?

Reply:
I guess there are two posts to the event loop until the state gets updated. Sleeping for 0.001s also worked, but Adminiuga suggested the multiple 0 sleeps instead.

Reply:
To give control back to the loop so it finishes the running tasks.

Reply:
@dmulcahey any chance you can merge it to the next milestone?
The ZHA groups behavior is quite annoying and seems like this fix should resolve the issue.

Thanks in advance!

Reply:
>why 2 of these back to back? this seems weird to me. is there something flaky in the tests that this is covering up?

It is because there's a sync function scheduling a task on the loop, whose status is not being tracked by HA so `hass.async_block_till_done()` does not take those tasks into consideration. If the test finishes before those tasks, then the test fails. `await sleep(0)` relinquishes control to the loop to give those tasks a chance to run. and finish.